### PR TITLE
scheduler: make `evict-slow-trend` scheduler can manually modify recovery duration.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What problem does this PR solve?

Issue Number: Ref #7156  , Ref https://github.com/tikv/tikv/issues/15271

### What is changed and how does it work?

```commit-message
With this pr, users can manually modify the minimal recovery time when encountering an I/O jitter case.

That is, only when the jitter is disappear and the recovery time reach this limit, 
can the given slow node be mark with normal for balancing leaders to it.
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/schedulers/evict_slow_trend.go [hunk 1]
```diff
@@ -55,10 +60,27 @@ type evictSlowTrendSchedulerConfig struct {
 	// Last chosen candidate for eviction.
 	lastEvictCandidate slowCandidate
 
+	RecoveryDurationGap uint64 `json:"recovery-duration"`
```
Comment (nolouch) [inline pkg/schedule/schedulers/evict_slow_trend.go:63]: add comments, and it's unit is secs.

**Review by nolouch — APPROVED** [review]
rest lgtm

**Review by nolouch — COMMENTED** [review]
need a reference to the issue.

**Comment by LykxSassinator** [comment]
> need a reference to the issue.

Got it, I'll supplement one issue for it.

File: pkg/schedule/schedulers/evict_slow_trend.go [hunk 2]
```diff
@@ -162,9 +193,50 @@ func (conf *evictSlowTrendSchedulerConfig) clearAndPersist(cluster sche.Schedule
...
+	router.HandleFunc("/list", h.ListConfig).Methods(http.MethodGet)
+	return router
+}
+
+func (handler *evictSlowTrendHandler) UpdateConfig(w http.ResponseWriter, r *http.Request) {
+	var input map[string]interface{}
+	if err := apiutil.ReadJSONRespondError(handler.rd, w, r.Body, &input); err != nil {
+		return
+	}
+	recoveryDurationGapFloat, ok := input["recovery-duration"].(float64)
```
Comment (bufferflies) [inline pkg/schedule/schedulers/evict_slow_trend.go:217]: why not convert to uint64 directly? BTW, Does the input needs to check also.

File: pkg/schedule/schedulers/evict_slow_trend.go [hunk 3]
```diff
@@ -162,9 +193,50 @@ func (conf *evictSlowTrendSchedulerConfig) clearAndPersist(cluster sche.Schedule
...
+	if ok {
+		recoveryDurationGap := (uint64)(recoveryDurationGapFloat)
+		prevRecoveryDurationGap := atomic.LoadUint64(&handler.config.RecoveryDurationGap)
+		atomic.StoreUint64(&handler.config.RecoveryDurationGap, recoveryDurationGap)
+		log.Info("evict-slow-trend-scheduler update 'recovery-duration' - unit: s", zap.Uint64("prev", prevRecoveryDurationGap), zap.Uint64("cur", recoveryDurationGap))
+	}
```
Comment (bufferflies) [inline pkg/schedule/schedulers/evict_slow_trend.go:223]: you should handler the `not ok` case. user may think this behavior  is successful but the config not update.

File: pkg/schedule/schedulers/evict_slow_trend.go (hunk 2 above)
Comment (LykxSassinator) [reply pkg/schedule/schedulers/evict_slow_trend.go:217]: IMO, the check is necessary. The given `body` might contain unexpected errors, which should be skipped.
And the conversion refers to:
https://github.com/tikv/pd/blob/edf1f95c2ef08dbc9af1c1f38a8ed5fdda469726/pkg/schedule/schedulers/evict_leader.go#L382.

**Review by bufferflies — APPROVED** [review]

**Pull request merged**